scipy/stats/_morestats.py

```diff
@@ -1673,7 +1674,18 @@ def _neg_llf(lmbda, data):
         return -llf
 
     with np.errstate(invalid='ignore'):
-        return optimize.brent(_neg_llf, brack=brack, args=(x,))
+        if brack is not None:
+            return optimize.brent(_neg_llf, brack=brack, args=(x,))
+        x = np.asarray(x)
+        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
+        b = np.log(np.finfo(dtype).eps)
+        max_x = 20 * np.nanmax(np.abs(x))
```

Comment:
1. Without this PR, the implementation fails with `scipy.optimize._optimize.BracketError: The algorithm terminated without finding a valid bracket. Consider trying different initial points.` when NaNs are present in the input data. Since this error is not helpful, I added an explicit check that fails with a `ValueError` when the input data is not finite. I also added tests to cover this case.
2. I replaced `np.nanmax` with `np.max` since there is now an explicit check for non-finite data.
3. Done.
4. Done.

Reply:
Thanks; all this is great.